compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -257,26 +250,21 @@ struct GPUSetEncodingOpLoweringConversion
     // the map_to_vector method here.
     auto encoding = IREE::Encoding::getEncodingAttr(encodingOp.getResultType());
     int64_t roleIdx = encoding.getOperandIndex().getInt();
-    auto elemTypes = llvm::map_to_vector(
-        encoding.getElementTypes().getValue(),
-        [](Attribute a) { return cast<TypeAttr>(a).getValue(); });
     auto loc = encodingOp.getLoc();
 
-    std::optional<TileMxNxK> intrinsicShape = getIntrinsicSize(elemTypes);
-    if (!intrinsicShape || intrinsicVectorShape.empty()) {
-      return failure();
-    }
-
     SmallVector<int64_t> targetShape; // for unrolling
     switch (roleIdx) {
     case 0: // A
-      targetShape = {intrinsicShape->M, intrinsicShape->K};
+      targetShape = {maybeEncodingInfo->intrinsicSize[0],
+                     maybeEncodingInfo->intrinsicSize[2]};
       break;
     case 1: // B
-      targetShape = {intrinsicShape->N, intrinsicShape->K};
+      targetShape = {maybeEncodingInfo->intrinsicSize[1],
+                     maybeEncodingInfo->intrinsicSize[2]};
       break;
     case 2: // C
-      targetShape = {intrinsicShape->M, intrinsicShape->N};
+      targetShape = {maybeEncodingInfo->intrinsicSize[0],
+                     maybeEncodingInfo->intrinsicSize[1]};
       break;
```

Comment:
What is the purpose of `targetShape`/`intrinsicSize`? I thought everything we need for the swizzling was in `innerTileSizes`, `innerTileShapes`, `permutation`.